ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -92,15 +105,54 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDefinitions = append(containerDefinitions, containerDef)
 	}
 
+	ecsParams, err := readECSParams(ecsParamsFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	var networkMode string
+	if ecsParams != nil {
+		networkMode = ecsParams.TaskDefinition.NetworkMode
+		// The task-role-arn flag should take precedence over a taskRoleArn value specified in ECS fields file
+		if taskRoleArn == "" {
+			taskRoleArn = ecsParams.TaskDefinition.TaskRoleArn
+		}
+	}
+
 	taskDefinition := &ecs.TaskDefinition{
 		Family:               aws.String(taskDefinitionName),
 		ContainerDefinitions: containerDefinitions,
 		Volumes:              convertToECSVolumes(volumes),
 		TaskRoleArn:          aws.String(taskRoleArn),
+		NetworkMode:          aws.String(networkMode),
 	}
+
 	return taskDefinition, nil
 }
 
+func readECSParams(filename string) (*ECSParams, error) {
+	if filename == "" {
+		defaultFilename := "ecs-params.yml"
+		if _, err := os.Stat(defaultFilename); err == nil {
+			filename = defaultFilename
+		} else {
+			return nil, nil
+		}
+	}
+	ecsParamsData, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error reading file '%v'", filename)
```

Comment:
How about something like "Error reading ECS params file: %s"

Reply:
@nrdlngr Could you take a look at the error messages and let me know what you recommend? Thanks!